File: backend-nestjs/src/modules/zones/domain/zone.repository.ts
import { Zone } from './zone';

export abstract class ZoneRepository {
  abstract findAll(): Promise<Zone[]>;
  abstract findById(id: string): Promise<Zone | null>;
  abstract save(zone: Zone): Promise<Zone>;
  abstract remove(id: string): Promise<void>;
  abstract existsByName(name: string): Promise<boolean>;
}

File: backend-springboot/src/main/java/com/greenalgeria/zone/domain/ZoneRepository.java
package com.greenalgeria.zone.domain;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface ZoneRepository {
    Optional<Zone> findById(UUID id);

    List<Zone> findAllByOrderByNameAsc();

    Zone save(Zone zone);

    boolean existsById(UUID id);

    void deleteById(UUID id);
}
